examples/face_recognition_11/face_recognition_11/seed_test_run.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import sys
+from argparse import ArgumentParser
+from argparse import Namespace
+from typing import List
+
+import pandas as pd
+from face_recognition_11.evaluator import evaluate_face_recognition_11
+from face_recognition_11.workflow import ThresholdConfiguration
+from face_recognition_11.workflow import Inference
+from face_recognition_11.workflow import Model
+from face_recognition_11.workflow import TestSample
+from face_recognition_11.workflow import TestSuite
+
+import kolena
+from kolena.workflow import test
+from kolena.workflow.annotation import BoundingBox
+from kolena.workflow.annotation import Keypoints
+
+BUCKET = "kolena-public-datasets"
+DATASET = "labeled-faces-in-the-wild"
+
+
+def seed_test_run(model_name: str, detector: str, test_suite_names: List[str]) -> None:
+    df = pd.read_csv(f"s3://{BUCKET}/{DATASET}/predictions/predictions_{model_name}.csv")
+
+    def infer(test_sample: TestSample) -> Inference:
+        similarities = []
+        for img in test_sample.pairs:
+            # assume no image pair with itself
+            query = df["locator_a"].isin([test_sample.locator, img.locator]) & df["locator_b"].isin(
+                [test_sample.locator, img.locator]
+            )
+            sample_results = df[query]
+            similarity = sample_results["similarity"].values[0] if not sample_results["failure"].values[0] else None
+            similarities.append(similarity)
+
+        if not df[df["locator_a"] == test_sample.locator].empty:
```

Comment:
I think we can clean this up by separating the detection data to a separate csv file where you can look it up simply by an image locator.